src/components/views/voip/CallView/CallViewButtons.tsx

```diff
@@ -56,11 +66,48 @@ const CallViewToggleButton: React.FC<IButtonProps> = ({ state: isOn, className,
     return (
         <AccessibleTooltipButton
             className={classes}
-            onClick={onClick}
             title={isOn ? onLabel : offLabel}
             alignment={Alignment.Top}
             yOffset={TOOLTIP_Y_OFFSET}
-        />
+            {...props}
+        >
+            { children }
+        </AccessibleTooltipButton>
+    );
+};
+
+interface IDropdownButtonProps extends IButtonProps {
+    deviceKinds: MediaDeviceKindEnum[];
+}
+
+const CallViewDropdownButton: React.FC<IDropdownButtonProps> = ({ state, deviceKinds, ...props }) => {
+    const [menuDisplayed, buttonRef, openMenu, closeMenu] = useContextMenu();
+    const [hoveringDropdown, setHoveringDropdown] = useState(false);
+
+    const classes = classNames("mx_CallViewButtons_button", "mx_CallViewButtons_dropdownButton", {
+        mx_CallViewButtons_dropdownButton_collapsed: !menuDisplayed,
+    });
+
+    const onClick = (event: React.MouseEvent): void => {
+        event.stopPropagation();
+        openMenu();
+    };
+
+    return (
+        <CallViewToggleButton inputRef={buttonRef} forceHide={menuDisplayed || hoveringDropdown} state={state} {...props}>
+            <CallViewToggleButton
+                className={classes}
+                onClick={onClick}
+                onHover={(hovering) => setHoveringDropdown(hovering)}
```

Comment:
Are you 100% sure that `onHover` takes the boolean state, instead of `event` type?

Reply:
Yes, see [`24960df` (#7173)](https://github.com/matrix-org/matrix-react-sdk/pull/7173/commits/24960dff9eccc0a53885f1d2decea125b2a2cb13)